theforeman.org/pipelines/test/testKatello.groovy

```diff
@@ -1,175 +1,187 @@
+                            dir('foreman') {
+                                configureDatabase(ruby)
+                            }
 
-            }
-        }
-        stage('Install Foreman npm packages') {
-            steps {
-                dir('foreman') {
-                    withRVM(["bundle exec npm install"], ruby)
-                }
-            }
-        }
-        stage('Run Tests') {
-            parallel {
-                stage('tests') {
-                    steps {
-                        dir('foreman') {
-                            withRVM(['bundle exec rake jenkins:katello TESTOPTS="-v" --trace'], ruby)
                         }
                     }
-                }
-                stage('rubocop') {
-                    steps {
-                        dir('foreman') {
-                            withRVM(['bundle exec rake katello:rubocop TESTOPTS="-v" --trace'], ruby)
+                    stage('Install Foreman npm packages') {
+                        steps {
+                            dir('foreman') {
+                                withRVM(["bundle exec npm install"], ruby)
+                            }
                         }
                     }
-                }
-                stage('react-ui') {
-                    when {
-                        expression { fileExists('package.json') }
+                    stage('rubocop') {
+                        steps {
+                            dir('foreman') {
+                                withRVM(['bundle exec rake katello:rubocop TESTOPTS="-v" --trace'], ruby)
+                            }
+                        }
                     }
-                    steps {
-                        sh "npm install"
-                        sh 'npm test'
+                    stage('react-ui') {
+                        when {
+                            expression { fileExists('package.json') }
+                        }
+                        steps {
+                            sh "npm install"
+                            sh 'npm test'
+                        }
                     }
-                }
-                stage('angular-ui') {
-                    steps {
-                        script {
-                            if (!fileExists('engines/bastion')) {
-                                dir('foreman') {
-                                    withRVM(['bundle show bastion > bastion-version'], ruby)
-
-                                    script {
-                                        bastion_install = readFile('bastion-version')
-                                        bastion_version = bastion_install.split('bastion-')[1]
-                                        echo bastion_install
-                                        echo bastion_version
+                    stage('angular-ui') {
+                        steps {
+                            script {
+                                if (!fileExists('engines/bastion')) {
```

Comment:
bastion was merged in 3.12, so it's safe to say this will always be "false" now

Reply:
Tackled.